### PR TITLE
add chloggen tool

### PR DESCRIPTION
**Pull request by codeboten**
This tool is used in the OpenTelemetry collector repositories to generate the changelog from a set of YAML files. The reason for moving this tool here is to make it available for both collector and collector contrib repos.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@codeboten lint failed on this.

**Comment by codeboten** [comment]
> LGTM, not sure how ownership works on this repo but maybe we want to copy the CODEOWNERS over from the contrib repo too?

added code owners from the contrib repo for chloggen

**Comment by codeboten** [comment]
> @codeboten lint failed on this.

@tigrannajaryan i ended up excluding the `testdata` directory from linting to work around the following two lint failures:

```
/app/testdata/all_change_types_multiple.md:46:81 MD013/line-length Line length [Expected: 80; Actual: 110]
/app/testdata/all_change_types_multiple.md:48 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "### 💡 Enhancements 💡"]
```

An alternative to excluding that directory would be to add a configuration in this repo for the linter. Any thoughts?

**Comment by codeboten** [comment]
@djaglowski PTAL as you're the code owner in the contrib repo and I'm adding you as a code owner in this repo for chloggen 😄

**Review by djaglowski — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@open-telemetry/go-maintainers please review.

**Comment by codeboten** [comment]
Any chance to get this merged? Otherwise we're gonna end up having to make changes in multiple places: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13121

File: chloggen/summary.go
Comment (MadVikingGod) [inline chloggen/summary.go:64]: Is this supposed to be baked into the binary?  As it is right now we will have to have a `summary.tmpl` file along side the binary that this is run from.

If it's an option, I would suggest using [`embed`](https://pkg.go.dev/embed) to get the template inside the binary.

File: chloggen/main.go
Comment (MadVikingGod) [inline chloggen/main.go:204]: nit: do we want to fall at the first error, or collect them all, so we have a list of what went wrong?

**Review by MadVikingGod — COMMENTED** [review]
LGTM

File: chloggen/summary.go
Comment (djaglowski) [reply chloggen/summary.go:64]: Yeah, some kind of configurability will be necessary. Even integrating this back into collector-contrib will force a change here, so I think we can expect a followup PR on this in the near term.

File: chloggen/main.go
Comment (djaglowski) [reply chloggen/main.go:204]: Collecting them all would be great. In practice it doesn't matter much as long as validation runs regularly. A PR will typically only add one entry, so there are a very small number of possible errors to work through.

**Comment by djaglowski** [comment]
Can this be merged?

**Review by tigrannajaryan — APPROVED** [review]
I haven't reviewed, but we have approvals from Go and Collector maintainers. Approving so that we can merge.

**Comment by tigrannajaryan** [comment]
FYI, this repo did not require linear commit history and this PR was merged with a merge commit. I have just enabled "Require linear history" for the future commits.

**Pull request merged**
